rfc/system/0000-flyte-agent.md

```diff
@@ -0,0 +1,141 @@
+
+While Flyte excels at orchestrating these relatively long-lived executions, the intrinsic overhead of singular k8s Pod execution is not well suited for low-latency workflows. In these scenarios, task executions are often measurable in milliseconds, and the overhead of cold-starting k8s Pods may be responsible for a large portion of the execution duration. Fortunately, the variance in task requirements (ex. resource requirements, dependencies, etc) tend to correlate with the execution durations -- meaning short-lived tasks are often quite similar. We hope to exploit this characteristic to more effectivley schedule low-latency workflows on Flyte by keeping k8s Pods "hot", so that a single instance may execute multiple tasks, mitigating the cold-start overhead with a new k8s Pod for each individual task execution.
+
+## 3 Proposed Implementation
+
+This document proposes a scheme where a collection of flyte agents are deployed providing a "hot" environment for task execution, mitigating the order of seconds overhead in managing individual k8s Pods per task. Because this paradigm is orthogonal to the current approach, there are a number of challenges in seamlessly integrating with Flyte, and these challenges are compounded with each step towards massive scale and optimized performance. Consequently, this solution is submitted as a lightweight, non-intrusive scheme to support low-latency workflows wihtin Flyte and should be treated as a PoC rather than an end solution. A key implementation component wll be profiling scale and performance improvements in overhead reduction.
+
+### Solution Architecture
+Existing solutions for similar problems may rely on distributed, persistent task queues. While, for massive scale especially, this is a proven solution -- it involves significant distributed architecture, furhter complicating FLyte integration and the dpeloyment model with additional dependencies. Our design is much more lightweight, leveraging existin Flyte components and constructs to emulate persistence, state consistency, Specifically we introduce two components, namely the Flyte Agent and Flyte Agent Plugin.
+
+**Flyte Agent**: The Flyte Agent is a Rust application that executes python tasks within the same application process to ensure minimal overhead. The [PyO3](TODO) library enables Rust to initialize Python environments to execute arbitrary code, in this scenario we envision the `pyflyte execute ...` command that Flyte currently uses to execute tasks in k8s Pods.
```

Comment:
nice, i've wanted to do this for a while, that's why i requested the rust buf entries be added.  want to make the i/o  portions of entrypoint.py in rust - and potentially see how a binary could operate as a pid 1 system init process as well.